apis/task.py

```diff
@@ -265,15 +262,134 @@ def post_process(self) -> DAGNode:
 
 
 @dataclasses.dataclass
-class GpuTask(BaseTask):
+class GpuCreateResourceTask(BaseTask):
   """This is a class to set up tasks for GPU.
 
   Attributes:
-    image: the image version that a GPU runs.
     image_project: the project that an image belongs to.
     image_family: the family group that an image belongs to.
   """
 
-  image: str
   image_project: str
   image_family: str
+  task_test_config: test_config.TestConfig[test_config.Gpu]
+  task_gcp_config: gcp_config.GCPConfig
+  task_metric_config: Optional[metric_config.MetricConfig] = None
+
+  def run(self) -> DAGNode:
+    """Run a test job.
+
+    Returns:
+      A task group with the following tasks chained: provision, run_model,
+      post_process.
+    """
+    # piz: We skip the queued resource for GPU for now since there is no queued resource command for GPU.
```

Comment:
what happens if the resource was unavailable? does it mean we have a higher chance of resource drop without queued resource?

Reply:
Yes, the request will be dropped after 3 attempts if resource is still unavailable. We should find a solution for queued resource for GPU in the future.